contracts/modules/assessment/Assessment.sol

```diff
@@ -2,515 +2,365 @@
-  ///                                 allows chunked processing. Can also be 0 if chunking is not
-  ///                                 necessary.
-  function processFraud(
-    uint256 rootIndex,
-    bytes32[] calldata proof,
-    address assessor,
-    uint256 lastFraudulentVoteIndex,
-    uint96 burnAmount,
-    uint16 fraudCount,
-    uint256 voteBatchSize
-  ) external override whenNotPaused {
-
-    require(
-      MerkleProof.verify(
-        proof,
-        fraudResolution[rootIndex],
-        keccak256(abi.encodePacked(assessor, lastFraudulentVoteIndex, burnAmount, fraudCount))
-      ),
-      InvalidMerkleProof()
-    );
-
-    Stake memory _stake = stakeOf[assessor];
-
-    // Make sure we don't burn beyond lastFraudulentVoteIndex
-    uint processUntil = _stake.rewardsWithdrawableFromIndex + voteBatchSize;
-
-    if (processUntil >= lastFraudulentVoteIndex) {
-      processUntil = lastFraudulentVoteIndex + 1;
+    if (voteSupport) {
+      assessment.acceptVotes++;
+    } else {
+      assessment.denyVotes++;
     }
 
-    for (uint j = _stake.rewardsWithdrawableFromIndex; j < processUntil; j++) {
-      IAssessment.Vote memory vote = votesOf[assessor][j];
-      IAssessment.Poll memory poll = assessments[vote.assessmentId].poll;
+    _assessments[claimId] = assessment;
 
-      {
-        if (uint32(block.timestamp) >= uint(poll.end) + PAYOUT_COOLDOWN) {
-          // Once the cooldown period ends the poll result is final, thus skip
-          continue;
-        }
-      }
+    _ballots[assessorMemberId][claimId] = Ballot({
+      timestamp: uint32(block.timestamp),
+      support: voteSupport
+    });
+    _ballotsMetadata[assessorMemberId][claimId] = ipfsHash;
 
-      if (vote.accepted) {
-        poll.accepted -= vote.stakedAmount;
-      } else {
-        poll.denied -= vote.stakedAmount;
-      }
+    emit VoteCast(claimId, msg.sender, assessorMemberId, voteSupport, ipfsHash);
+  }
 
-      // If the poll ends in less than 24h, extend it to 24h
-      uint32 nextDay = uint32(block.timestamp + 1 days);
-      if (poll.end < nextDay) {
-        poll.end = nextDay;
+  /// @notice Allows for the early closing of a claim's voting period.
+  /// @dev Can only be called if all assigned assessors have cast their votes.
+  ///      Must be called by an active assessor for the claim to prevent potential front-running.
+  ///      Sets the assessment's `votingEnd` to the current block timestamp.
+  /// @param claimId The unique identifier for the claim.
+  function closeVotingEarly(uint claimId) override external {
+    // an assessor on the claim must call this to prevent front-running
+    (, Assessment memory assessment) = _validateAssessor(claimId, msg.sender);
+    require(block.timestamp < assessment.votingEnd, VotingAlreadyClosed());
+
+    uint[] memory assessors = getGroupAssessors(assessment.assessingGroupId);
+    uint groupSize = assessors.length;
```

Comment:
you should use the set's length instead of fetching all members from storage

Reply:
so we want to use the storage approach instead of memory on the loop when checking if a current assessor has voted or not.

with memory, at best it will be pretty much similar to storage if all have voted, but we might save a bit of gas if we use storage and not all have voted.